awx/main/tasks.py

```diff
@@ -354,6 +355,22 @@ def send_notifications(notification_list, job_id=None):
 
 @task(queue=get_local_queuename)
 def gather_analytics():
+    def _gather_and_ship(subset, since=None, until=None):
+        try:
+            tgz = analytics.gather(subset=subset, since=since, until=until)
+            # empty analytics without raising an exception is not an error
+            if not tgz:
+                return True
+            logger.info('gathered analytics: {}'.format(tgz))
+            analytics.ship(tgz)
```

Comment:
hmm, so one new issue and one old issue

new issue
-------------

https://github.com/ansible/awx/blob/devel/awx/main/analytics/core.py#L186-L189 will save the last updated timestamp for each collection, so if this fails in the middle, we'll miss a lot of data

So we should save the state.last_run after we've sent all the bundles, right?

And old issue
-------------------

It gets fresh `now()` before shipping, so this creates a gap now?

E.g. I'll be collecting events where the `now()` is `2020-07-01T12:00:00`, but this will take a while and when I want to ship this I take another `now()` at. e.g. `2020-07-01T13:00:00` and say this is the last run

That will create 1h of gap a data we'll never send to analytics. Should I create separate issue for this?

Basically, we should take `now()` snapshot at the start and work with it for all collections and set it as `state.last_run` after all payloads were sent.

Reply:
The solo model there shouldn't actually used now that this PR adds the `since` and `until` parameters to gather() and always passes them.
 
It probably should be removed for clarity.